// @infino/data-models/ancillary/seeds/20200421063629-collection.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'collections', [
        {
            id: '9f81b97e-839b-11ea-bc55-0242ac130003',
            name: 'Winter Collection',
            description: 'The best of the winter collection',
            image_url: 'http://colection-image.com',
            color: '#FFF111',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            name: 'Test Collection 1',
            description: 'Here we\'ll list the offers for collection 1',
            image_url: 'https://i.ya-webdesign.com/images/pants-clipart-transparent-background-3.png',
            color: '#259CC9',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'b8993a37-d9e3-430e-a132-7c6096e40601',
            name: 'Test Collection 2',
            description: 'Here we\'ll list the offers for collection 2',
            image_url: 'http://www.pngmart.com/files/1/Nike-Shoes-Transparent-Background.png',
            color: '#75270D',
            created_at: new Date(),
            updated_at: new Date()
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'collections', null, {} ),

};
