// @infino/data-models/ancillary/seeds/20200606082809-online-offer-finance.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'online_offer_finances', [
        {
            id: 'a8fa7de3-8d77-4b3e-9c0e-4146b1710a23',
            offer_id: '7fb4ace6-1420-46ef-b3bd-9e065c4e79d0',
            finance_id: 'a9c39d32-a0d9-4180-a55b-c634d9f6f38f',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'edb9c9a5-1a3b-4e08-baa1-6ab8e02d0f4e',
            offer_id: '8ccf3502-81e1-4a5a-a52a-512dfc99dcff',
            finance_id: '01e89053-1684-4f03-bd8d-7553c5542465',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '3eb47353-2e7d-4bbc-980c-0880221a5e35',
            offer_id: 'd147263f-6507-4a96-97c8-4766ebc4bc0a',
            finance_id: 'c03f9c22-7f8e-48af-8c3b-5b4580394896',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '500af739-7674-44e4-a839-607f030b4c8e',
            offer_id: '073c63e8-7175-46d0-adeb-8387f17029fd',
            finance_id: '60868d0f-5e0a-4c41-8a29-2c2a7dfb90ee',
            created_at: new Date(),
            updated_at: new Date(),
        },
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'online_offer_finances', null, {} ),

};
