// @infino/data-models/ancillary/seeds/20200421065652-online-merchant.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'online_merchants', [

        {
            id: 'f362ac15-0b4a-4739-8984-094277597515',
            name: 'AliExpress',
            merchant_url: 'https://www.aliexpress.com/',
            android_package_identifier: 'android_ali',
            ios_package_identifier: '',
            logo_url: 'https://logodix.com/logo/43067.png',
            cover_image_url: 'https://images.unsplash.com/photo-1523554888454-84137e72c3ce?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
            host_name: 'flipkart.com',
            background_color: '#FFFFFF',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'e2091a1c-c1fd-44bc-b722-9f072cd74109',
            name: 'Amazon',
            merchant_url: 'https://www.amazon.in',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: 'https://logodix.com/logo/233716.jpg',
            cover_image_url: 'https://images.unsplash.com/photo-1545050814-0a22f900193e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
            host_name: 'flipkart.com',
            background_color: '#A15FFF',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '2323da38-0288-4223-9967-8b1ece39a618',
            name: 'Flipkart',
            merchant_url: 'https://www.flipkart.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: '',
            cover_image_url: 'https://images.unsplash.com/photo-1547891654-e66ed7ebb968?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1350&q=80',
            host_name: 'flipkart.com',
            background_color: '#3455FF',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'aafe2eb0-3269-471b-880b-cb9ac81dba15',
            name: 'Booking.com',
            merchant_url: 'https://www.booking.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: 'https://cdn.worldvectorlogo.com/logos/bookingcom-1.svg',
            cover_image_url: 'https://images.unsplash.com/photo-1572450732467-5eb1311e7bc8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80',
            host_name: 'booking.com',
            background_color: '#EA4455',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'da4d0024-130e-40a2-b56b-f372608c20ce',
            name: 'Snapdeal',
            merchant_url: 'https://www.snapdeal.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: 'https://lh3.googleusercontent.com/IfMz_US0gMwcVYtNNjQPEglG4cHLW51zitBhGZ0nmAYEXTIuIdCjUaZ_Xg9H6sDVxQ=s360-rw',
            cover_image_url: 'https://images.unsplash.com/photo-1511424443513-a00662140eeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1349&q=80',
            host_name: 'snapdeal.com',
            background_color: '#234BB5',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '8e94b15f-e212-47e3-94e4-d6ec580d09b6',
            name: 'GoDaddy',
            merchant_url: 'https://www.godaddy.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: 'https://logodix.com/logo/1883881.png',
            cover_image_url: 'https://images.unsplash.com/photo-1542562401-cee87aaed85d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80',
            host_name: 'flipkart.com',
            background_color: '#234BB5',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '1fad6fd2-b2bc-4c03-a18f-b0ceef02bac6',
            name: 'Paytm',
            merchant_url: 'https://www.paytm.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: 'https://logodix.com/logo/354316.png',
            cover_image_url: 'https://images.unsplash.com/photo-1544840413-6d9583494530?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80',
            host_name: 'flipkart.com',
            background_color: '#934BB5',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '2eb0f80d-54be-4fb2-bca3-c0ecf39db62d',
            name: 'EspnCricInfoShop',
            merchant_url: 'https://in.shop.espncricinfo.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: 'https://www.seekpng.com/png/full/31-310147_espncricinfo-mobile-logo-design-apps-espncricinfo-logo-png.png',
            cover_image_url: 'https://images.unsplash.com/photo-1560188305-788b6818a21a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80',
            host_name: 'flipkart.com',
            background_color: '#634BB5',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'd340fe88-c689-4198-bf1f-0e646916d41e',
            name: 'Rediff Shopping',
            merchant_url: 'https://shopping.rediff.com',
            android_package_identifier: '',
            ios_package_identifier: '',
            logo_url: '',
            cover_image_url: 'https://images.unsplash.com/photo-1554397841-1ca3f894b6d0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1262&q=80',
            host_name: 'flipkart.com',
            background_color: '#456BB5',
            created_at: new Date(),
            updated_at: new Date(),
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'online_merchants', null, {} ),

};
