// @infino/data-models/ancillary/seeds/20200421065705-online-offer.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'online_offers', [
        {
            id: '7fb4ace6-1420-46ef-b3bd-9e065c4e79d0',
            merchant_id: '2323da38-0288-4223-9967-8b1ece39a618',
            offer_text: 'Get Best Deals',
            validity: '2021-01-08',
            applicability: 'selected stores',
            coupon_code: 'MAKEBA',
            conditions: [
                'first condition',
                'second condition'
            ],
            infino_conditions: [
                'first infino condition',
                'second infino condition'
            ],
            redemption_link: 'http://redd-link-1.com',
            description: 'A great offer',
            offer_color: '#FFAA63',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '8ccf3502-81e1-4a5a-a52a-512dfc99dcff',
            merchant_id: '2323da38-0288-4223-9967-8b1ece39a618',
            offer_text: 'Get Best Deals',
            validity: '2021-01-08',
            applicability: 'selected stores',
            coupon_code: 'TOP100',
            conditions: [
                'first condition',
                'second condition'
            ],
            infino_conditions: [
                'my infino condition',
                'your infino condition'
            ],
            redemption_link: 'http://redd-link-2.com',
            description: 'A great offer',
            offer_color: '#FFAA63',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'd147263f-6507-4a96-97c8-4766ebc4bc0a',
            merchant_id: '2323da38-0288-4223-9967-8b1ece39a618',
            offer_text: 'Get Best Deals',
            validity: '2021-01-08',
            applicability: 'selected stores',
            coupon_code: 'MAKEBA',
            conditions: [
                'first condition',
                'second condition'
            ],
            infino_conditions: [
                'a infino condition',
                'another infino condition'
            ],
            description: 'A great offer',
            redemption_link: 'http://redd-link-3.com',
            offer_color: '#FFAA63',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '073c63e8-7175-46d0-adeb-8387f17029fd',
            merchant_id: 'aafe2eb0-3269-471b-880b-cb9ac81dba15',
            offer_text: 'Get Best Deals',
            validity: '2021-01-08',
            applicability: 'selected stores',
            coupon_code: 'MAKEBA',
            conditions: [
                'first condition',
                'second condition'
            ],
            infino_conditions: [
                'good infino condition',
                'great infino condition'
            ],
            description: 'A great offer',
            redemption_link: 'http://redd-link-4.com',
            offer_color: '#FFAA63',
            created_at: new Date(),
            updated_at: new Date(),
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'online_offers', null, {} ),

};
