// @infino/data-models/ancillary/seeds/20200519131802-card.js
'use strict';

module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'cards', [
        {
            id: '4ad9e5ec-1bdb-448a-ace9-32fb4716df9e',
            card_type_id: '9e9db538-8398-11ea-bc55-0242ac130003',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            name: 'Test First',
            first_six: '894204',
            last_four: '5793',
            card_token: '943967ff-4d63-4700-958b-bef47dd7809v',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '4ad9e5ec-1bdb-448a-ace9-32fb4716df7f',
            card_type_id: 'ef54ae3c-b7fb-4f4c-8f12-836227d1dfec',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            name: 'Test Second',
            first_six: '730029',
            last_four: '2830',
            card_token: '943967ff-4d63-4700-958b-bef47dd6701j',
            created_at: new Date(),
            updated_at: new Date(),
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'cards', null, {} ),

};
