// @infino/data-models/ancillary/seeds/20200501141702-order.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'orders', [
        {
            id: '58e4b8e0-1967-468e-bf14-fab33517fc84',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            payment_method_id: '4dbfa547-3a66-42dc-a4ea-94ec2064f4d1',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            giftcard_id: 'da1e653b-7b7f-4739-981a-cdcdcd87fdb4',
            amount: 500.99,
            profit_amount: 522.22,
            profit_percentage: 5.6,
            payable_amount: 300.12,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '4520ae87-d983-48e5-87d0-76823ad82a05',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            payment_method_id: '10d2f4fc-042f-4a94-b8bf-63aba18fe1f3',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            giftcard_id: '70286eb4-395a-4f45-a1e4-2682a430892c',
            amount: 500.99,
            profit_amount: 522.22,
            profit_percentage: 5.6,
            payable_amount: 300.12,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '0f7783b8-84f0-448d-b6db-406b55a65310',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            payment_method_id: '4dbfa547-3a66-42dc-a4ea-94ec2064f4d1',
            provider_id: 'ae5246af-2aef-4d43-bea9-e4bf442f3d9b',
            giftcard_id: '44a578dd-7642-472e-995a-5e845ee0ddec',
            amount: 500.99,
            profit_amount: 522.22,
            profit_percentage: 5.6,
            payable_amount: 300.12,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'b129805e-b0ab-4d28-9a73-4591de5a3d7e',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            payment_method_id: '10d2f4fc-042f-4a94-b8bf-63aba18fe1f3',
            provider_id: 'ae5246af-2aef-4d43-bea9-e4bf442f3d9b',
            giftcard_id: '14c19394-b5ae-4497-9399-9d16068d48f3',
            amount: 500.99,
            profit_amount: 522.22,
            profit_percentage: 5.6,
            payable_amount: 300.12,
            created_at: new Date(),
            updated_at: new Date()
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'orders', null, {} )

};
