// @infino/data-models/ancillary/seeds/20200606074638-finance.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'finances', [
        {
            id: 'a9c39d32-a0d9-4180-a55b-c634d9f6f38f',
            name: 'HDFC Offer',
            logo_url: 'url for logo',
            color: '#FFAA63',
            type: 'BANK',
            quick_filter: true,
            host_name: 'http://flipkart.com',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '01e89053-1684-4f03-bd8d-7553c5542465',
            name: 'ICICI Offer',
            bank_id: '9e9daf2a-8398-11ea-bc55-0242ac130003',
            logo_url: 'url for logo',
            color: '#FFAA63',
            type: 'FINTECH',
            quick_filter: true,
            host_name: 'hostname url',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'c03f9c22-7f8e-48af-8c3b-5b4580394896',
            name: 'Kotak Offer',
            bank_id: '3a3efe49-4d67-421f-8937-de0f750ca3e7',
            logo_url: 'url for logo',
            color: '#FFAA63',
            type: 'NETWORK',
            quick_filter: true,
            host_name: 'url hostname',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '60868d0f-5e0a-4c41-8a29-2c2a7dfb90ee',
            name: 'HSBC Offer',
            logo_url: 'url for logo',
            color: '#FFAA63',
            type: 'BANK',
            quick_filter: true,
            host_name: 'hostname url',
            created_at: new Date(),
            updated_at: new Date(),
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'finances', null, {} ),

};
