// @infino/data-models/ancillary/seeds/20200429111931-giftcard.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'giftcards', [
        {
            id: 'da1e653b-7b7f-4739-981a-cdcdcd87fdb4',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            merchant_id: 'f362ac15-0b4a-4739-8984-094277597515',
            giftcard_name: 'Giftcard 1',
            max_clubbable_quantity: 1,
            base_commission: 4.5,
            selling_rate: 4.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'GYFTR_SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '44a578dd-7642-472e-995a-5e845ee0ddec',
            provider_id: 'ae5246af-2aef-4d43-bea9-e4bf442f3d9b',
            merchant_id: 'f362ac15-0b4a-4739-8984-094277597515',
            giftcard_name: 'Giftcard 2',
            max_clubbable_quantity: 1,
            base_commission: 4.5,
            selling_rate: 8.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '14c19394-b5ae-4497-9399-9d16068d48f3',
            provider_id: 'ae5246af-2aef-4d43-bea9-e4bf442f3d9b',
            merchant_id: 'aafe2eb0-3269-471b-880b-cb9ac81dba15',
            giftcard_name: 'Giftcard 3',
            max_clubbable_quantity: 10,
            base_commission: 3.5,
            selling_rate: 9.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'RANGE',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '70286eb4-395a-4f45-a1e4-2682a430892c',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            merchant_id: 'da4d0024-130e-40a2-b56b-f372608c20ce',
            giftcard_name: 'Giftcard 4',
            max_clubbable_quantity: 1,
            base_commission: 4.5,
            selling_rate: 4.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'GYFTR_SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '5aabdfa9-a059-448e-90e1-398ef6e89b44',
            provider_id: 'ae5246af-2aef-4d43-bea9-e4bf442f3d9b',
            merchant_id: 'aafe2eb0-3269-471b-880b-cb9ac81dba15',
            giftcard_name: 'Giftcard 4',
            max_clubbable_quantity: 1,
            base_commission: 10,
            selling_rate: 1.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'RANGE',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '4d7cf7e9-4880-45f6-a138-b6ad0ec68b72',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            merchant_id: 'd340fe88-c689-4198-bf1f-0e646916d41e',
            giftcard_name: 'Archies',
            max_clubbable_quantity: 1,
            base_commission: 10,
            selling_rate: 1.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'GYFTR_SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '56d0c81a-88d6-45f8-8a2c-545011890534',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            merchant_id: 'd340fe88-c689-4198-bf1f-0e646916d41e',
            giftcard_name: 'Baskin Robbins',
            max_clubbable_quantity: 1,
            base_commission: 10,
            selling_rate: 1.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'GYFTR_SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '1b4837c9-72d1-4ea4-bd61-f78587d62762',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            merchant_id: 'd340fe88-c689-4198-bf1f-0e646916d41e',
            giftcard_name: 'Bata',
            max_clubbable_quantity: 1,
            base_commission: 10,
            selling_rate: 1.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'GYFTR_SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '5b22d8c2-d851-4e2b-8ebf-3f4d2320445a',
            provider_id: '6b038d03-a0f0-4414-bafa-f0ad9879b0d7',
            merchant_id: 'd340fe88-c689-4198-bf1f-0e646916d41e',
            giftcard_name: 'Croma',
            max_clubbable_quantity: 1,
            base_commission: 10,
            selling_rate: 1.5,
            about: 'Get an amazing Discount',
            expiry: '6 months',
            redemption_instructions: 'Put it on the homepage',
            applicability: 'all products',
            redemption_link: 'localhost',
            sale_applicability: true,
            exclusive_online: false,
            exclusive_offline: false,
            tnc_url: 'test.com',
            denomination_type: 'GYFTR_SLAB',
            created_at: new Date(),
            updated_at: new Date()
        },
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'giftcards', null, {} )

};
