// @infino/data-models/ancillary/seeds/20200501142013-mobikwik-payment-response.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'mobikwik_payment_responses', [
        {
            id: '94e2de37-30fe-4535-8706-44f2dd6f525a',
            order_id: '58e4b8e0-1967-468e-bf14-fab33517fc84',
            response_code: '200',
            gateway_transaction_id: 'test',
            gateway_transaction_time: new Date(),
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'b30bb42b-a2aa-4862-9e8b-0437ea7801a5',
            order_id: '4520ae87-d983-48e5-87d0-76823ad82a05',
            response_code: '200',
            gateway_transaction_id: 'test',
            gateway_transaction_time: new Date(),
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '89e9ec8a-bb51-4238-93d5-f9ffc157f1a9',
            order_id: '0f7783b8-84f0-448d-b6db-406b55a65310',
            response_code: '200',
            gateway_transaction_id: 'test',
            gateway_transaction_time: new Date(),
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '52124736-392a-41b0-9e42-e0855dc7661d',
            order_id: 'b129805e-b0ab-4d28-9a73-4591de5a3d7e',
            response_code: '200',
            gateway_transaction_id: 'test',
            gateway_transaction_time: new Date(),
            created_at: new Date(),
            updated_at: new Date()
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'mobikwik_payment_responses', null, {} )

};
