// @infino/data-models/ancillary/seeds/20200421065225-tag.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'tags', [

        {
            id: '81736372-d1a3-4be9-91f5-35393b3a9cfb',
            name: 'Electronics',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'c2c9fa15-9383-479f-946e-1576998a5375',
            name: 'Travel',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'e25a0a8f-a57b-4c7a-8025-a1a674c9db36',
            name: 'Shopping',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '5387771f-ee68-4f38-902c-4edc47706e69',
            name: 'Entertainment',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'b5f31f27-305a-4815-b946-9ca0d831b501',
            name: 'Essentials',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '5c441b7d-3459-48ca-860d-09373ab5676f',
            name: 'Food',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'd0b862b1-d9e6-414a-9f0d-a1676024f3e5',
            name: 'Health',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '84419036-8f45-4e44-9b4a-ac06487fa9c2',
            name: 'Business',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '32fafea6-0ff6-4cdc-a63a-e3e0802cc308',
            name: 'Clothing',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '352c14d1-3c82-4aab-ab11-2e3cf1831a0f',
            name: 'Education',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '1cc90def-79f8-417b-a86c-f0e766748724',
            name: 'Jewellery',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '76eface2-37f5-41eb-8220-3b4bb659b39d',
            name: 'Fitness',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '4d09db12-1f6d-4a95-907b-4204506daf89',
            name: 'Insurance',
            created_at: new Date(),
            updated_at: new Date(),
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'tags', null, {} )

};
