// @infino/data-models/ancillary/seeds/20200503123539-purchased-giftcard.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'purchased_giftcards', [
        {
            id: 'dda58461-a721-44de-8ec2-ceb54164fd3e',
            payment_response_reference: '94e2de37-30fe-4535-8706-44f2dd6f525a',
            provider_reference: '1571119071937',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            card_number: '4700001230064884',
            card_pin: 'LZDGIFT479427',
            activation_url: 'https://{baseUrl}/claim/470000/a/7SfzYJxRusNzctUk',
            activation_code: 'test code',
            expires_at: '2020-04-11T18:30:00+00:00',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '8c55c3fd-363b-449c-864c-8bc95fcc7245',
            payment_response_reference: 'b30bb42b-a2aa-4862-9e8b-0437ea7801a5',
            provider_reference: '1571119071937',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            card_number: '4700001230064884',
            zk_card_reference: 'test zkCardReference',
            expires_at: '2020-04-11T18:30:00+00:00',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'ab161e15-5ce5-4850-8da3-a27acee061b1',
            payment_response_reference: '89e9ec8a-bb51-4238-93d5-f9ffc157f1a9',
            provider_reference: '1571119071937',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            card_number: '4700001230064884',
            activation_url: 'https://{baseUrl}/claim/470000/a/7SfzYJxRusNzctUk',
            expires_at: '2020-04-11T18:30:00+00:00',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '97543249-f3cf-4a27-a90d-3216ac3fcf9a',
            payment_response_reference: '52124736-392a-41b0-9e42-e0855dc7661d',
            provider_reference: '1571119071937',
            user_id: '943967ff-4d63-4700-958b-bef47dd5836c',
            card_number: '4700001230064884',
            activation_code: 'test code',
            expires_at: '2020-04-11T18:30:00+00:00',
            created_at: new Date(),
            updated_at: new Date()
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'purchased_giftcards', null, {} )

};
