// @infino/data-models/ancillary/seeds/20200421063644-curated-offer.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'curated_offers', [
        {
            id: '4fa842c0-839f-11ea-bc55-0242ac130003',
            collection_id: '9f81b97e-839b-11ea-bc55-0242ac130003',
            category: 'Clothing',
            offer_title: 'Get 20% OFF',
            offer_discount: '20% FLAT OFF',
            description: 'On all sales you can get a flat discount, hurry!',
            merchant_name: 'H&M',
            merchant_logo_url: 'https://cdn-static-1.medium.com/_/fp/icons/favicon-rebrand-medium.3Y6xpZ-0FSdWDnPM3hSBIA.ico',
            merchant_bg_url: 'https://bg.medium.com/_/fp/icons/a-typical-background.ico',
            offer_image_url: 'https://image.com/img.png',
            listview_color: '#5fe4e1',
            detailview_color: '#a164ee',
            validity: '2021-01-08',
            applicability: 'all products',
            coupon_code: 'ALLADIN',
            conditions: [
                'the first condition',
                'another condition'
            ],
            redemption_link: 'https://redeem.com/',
            main_tnc: 'This is the primary condition.',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            collection_id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            category: 'Food',
            offer_discount: '30% off',
            offer_title: 'The tea that makes you go Badaaaamn!',
            description: '20% Discount for Orders above Rs 500.',
            merchant_name: 'Vahdham Teas',
            merchant_logo_url: 'https://i.imgur.com/STlNDTQ.png',
            merchant_bg_url: 'https://images.unsplash.com/photo-1523554888454-84137e72c3ce?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
            offer_image_url: 'https://i.imgur.com/0irt3kR.png',
            listview_color: '#115740',
            detailview_color: '#3FCB9D',
            validity: '05/30/2020',
            applicability: '',
            coupon_code: 'DONUT30',
            conditions: [
                'Maximum 1 use per customer. Standard Shipping & COD charges apply'
            ],
            main_tnc: 'This coupon cannot be used on top of other offers on the merchant',
            redemption_link: 'https://www.vahdamteas.com/',
            id: 'da389d79-939c-4271-9ea1-ae6789f69745',
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            collection_id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            category: 'Shopping',
            offer_discount: '30% off',
            offer_title: 'For everyone loves healthy, shiny & fuller hair.',
            description: 'Buy for minimum INR 999 and get Flat INR 300/- Off',
            merchant_name: 'Mamaearth',
            merchant_logo_url: 'https://i.imgur.com/6U6BQZa.png',
            merchant_bg_url: 'https://images.unsplash.com/photo-1545050814-0a22f900193e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80',
            offer_image_url: 'https://i.imgur.com/b0WJR5U.png',
            listview_color: '#FFC3CD',
            detailview_color: '#FFC3CD',
            validity: '05/30/2020',
            applicability: '',
            coupon_code: 'DONUT30',
            conditions: [
                'Offer cannot be clubbed with any other offers and is not transferable.',
                'Partial redemption of the coupon is not allowed',
                'Order placed using these coupons can\'t be modified',
                'Not applicable on KITS & TUFF'
            ],
            main_tnc: 'This coupon cannot be used on top of other offers on the merchant',
            redemption_link: 'https://mamaearth.in/',
            id: 'e57723c2-f81c-40ae-b0aa-d7f59e7efa5f',
            created_at: new Date(),
            updated_at: new Date()
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'curated_offers', null, {} ),

};
