// @infino/data-models/ancillary/seeds/20200430142851-gyftr-slab.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'gyftr_slabs', [
        {
            id: '68586dd8-9f99-445f-86ad-55cc0661fb37',
            giftcard_id: 'da1e653b-7b7f-4739-981a-cdcdcd87fdb4',
            amount: 200,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '3921303a-3263-48ad-bb0a-c7eeb7d46cf4',
            giftcard_id: '70286eb4-395a-4f45-a1e4-2682a430892c',
            amount: 100,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'B360368F-F4DE-475A-B718-23F2BCFF35AE',
            giftcard_id: '4d7cf7e9-4880-45f6-a138-b6ad0ec68b72',
            amount: 100,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '7B25A807-D54D-4B59-BF33-E225E0586039',
            giftcard_id: '56d0c81a-88d6-45f8-8a2c-545011890534',
            amount: 100,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '38C4C1F1-94EB-4666-A001-F4D87E5FD5E7',
            giftcard_id: '56d0c81a-88d6-45f8-8a2c-545011890534',
            amount: 50,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: '9762FF22-15CE-415A-89BC-6CA4DFA26300',
            giftcard_id: '1b4837c9-72d1-4ea4-bd61-f78587d62762',
            amount: 100,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'E8336DBE-2AB1-4B09-88A6-10F1FCB7A69C',
            giftcard_id: '1b4837c9-72d1-4ea4-bd61-f78587d62762',
            amount: 500,
            created_at: new Date(),
            updated_at: new Date()
        },
        {
            id: 'B829A455-8BB1-40C5-A7CE-3D7EB5457172',
            giftcard_id: '5b22d8c2-d851-4e2b-8ebf-3f4d2320445a',
            amount: 100,
            created_at: new Date(),
            updated_at: new Date()
        },
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'gyftr_slabs', null, {} )

};
