// @infino/data-models/ancillary/seeds/20200503173801-curated-giftcard.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'curated_giftcards', [
        {
            id: '4fa842c0-839f-11ea-bc55-0242ac130003',
            collection_id: '9f81b97e-839b-11ea-bc55-0242ac130003',
            giftcard_id: '14c19394-b5ae-4497-9399-9d16068d48f3',
            merchant_logo_url: 'https://image.com/img.png',
            category: 'Food',
            offer_discount: 'Get 30% OFF on all SWIGY orders',
            listview_color: '#a164ee',
            offer_image_url: 'https://image.com/img.png',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            collection_id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            giftcard_id: '70286eb4-395a-4f45-a1e4-2682a430892c',
            merchant_logo_url: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/H%26M-Logo.svg/1024px-H%26M-Logo.svg.png',
            category: 'Clothing',
            offer_discount: '50% Off',
            listview_color: '#FF9CAD',
            offer_image_url: 'https://purepng.com/public/uploads/medium/purepng.com-mens-original-jeansgarmentlower-bodydenimjeansblue-1421526363746aedsz.png',
            id: 'fb1b68f8-3652-4df8-848f-1fd94341e162',
            created_at: '2020-05-22T11:01:33.370Z',
            updated_at: '2020-05-22T11:01:33.370Z'
        },
        {
            collection_id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            giftcard_id: '14c19394-b5ae-4497-9399-9d16068d48f3',
            merchant_logo_url: 'https://1000logos.net/wp-content/uploads/2017/03/McDonalds-Logo.png',
            category: 'Food',
            offer_discount: '25% Off',
            listview_color: '#FDC82F',
            offer_image_url: 'https://www.freepnglogos.com/uploads/food-png/fast-food-transparent-png-pictures-icons-and-png-17.png',
            id: 'aa1e540d-f17c-4fbe-b7ab-5d4b9d9f9386',
            created_at: '2020-05-22T11:01:33.370Z',
            updated_at: '2020-05-22T11:01:33.370Z'
        },
        {
            collection_id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            giftcard_id: '44a578dd-7642-472e-995a-5e845ee0ddec',
            merchant_logo_url: 'https://i.imgur.com/n2G1lGM.png',
            category: 'Shoes',
            offer_discount: '30% Off',
            listview_color: '#000000',
            offer_image_url: 'http://clipart-library.com/images_k/shoes-transparent/shoes-transparent-7.png',
            id: 'f34f2f26-9582-40df-9d1b-e5dd7d1dc092',
            created_at: '2020-05-22T11:01:33.370Z',
            updated_at: '2020-05-22T11:01:33.370Z'
        },
        {
            collection_id: 'e7500f53-a31a-44d0-ac90-f5f5def83fd4',
            giftcard_id: 'da1e653b-7b7f-4739-981a-cdcdcd87fdb4',
            merchant_logo_url: 'https://i.imgur.com/KVUUBBL.png',
            category: 'Tacos',
            offer_discount: '20% Off',
            listview_color: '#D7BFFA',
            offer_image_url: 'https://purepng.com/public/uploads/medium/purepng.com-mens-original-jeansgarmentlower-bodydenimjeansblue-1421526363746aedsz.png',
            id: '892ad328-2412-4216-aa99-61dce2aa22f4',
            created_at: '2020-05-22T11:01:33.370Z',
            updated_at: '2020-05-22T11:01:33.370Z'
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'curated_giftcards', null, {} ),

};
