// @infino/data-models/ancillary/seeds/20200421084522-online-merchant-tag.js
module.exports = {

    up: queryInterface => queryInterface.bulkInsert( 'online_merchant_tags', [
        {
            id: 'ccfd53d3-5490-4f1e-920f-4096ca8e16c6',
            online_merchant_id: '2323da38-0288-4223-9967-8b1ece39a618',
            tag_id: '81736372-d1a3-4be9-91f5-35393b3a9cfb',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'ac66936a-5e5a-49bd-b286-1fdc1495bd62',
            online_merchant_id: '2323da38-0288-4223-9967-8b1ece39a618',
            tag_id: 'd0b862b1-d9e6-414a-9f0d-a1676024f3e5',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '9bf2757f-1d41-4fb7-ad3a-314ae5d43041',
            online_merchant_id: 'da4d0024-130e-40a2-b56b-f372608c20ce',
            tag_id: 'd0b862b1-d9e6-414a-9f0d-a1676024f3e5',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: '687165fb-a558-4c08-89ee-4e4c65a2ddb9',
            online_merchant_id: 'aafe2eb0-3269-471b-880b-cb9ac81dba15',
            tag_id: '81736372-d1a3-4be9-91f5-35393b3a9cfb',
            created_at: new Date(),
            updated_at: new Date(),
        },
        {
            id: 'faa6a9ff-1caf-4035-a49a-7bba0b6f3493',
            online_merchant_id: 'aafe2eb0-3269-471b-880b-cb9ac81dba15',
            tag_id: '84419036-8f45-4e44-9b4a-ac06487fa9c2',
            created_at: new Date(),
            updated_at: new Date(),
        }
    ], {} ),

    down: queryInterface => queryInterface.bulkDelete( 'online_merchant_tags', null, {} )

};
