// @infino/data-models/lib/postgres/schema/card.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    cardTypeId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'card_types',
            key: 'id',
        },
    },

    userId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'users',
            key: 'id',
        },
    },

    name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    firstSix: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    lastFour: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    cardToken: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    isActive: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: true,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
