// @infino/data-models/lib/postgres/schema/curated-offer.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    collectionId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'collections',
            key: 'id',
        },
    },

    category: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    offerDiscount: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    offerTitle: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    description: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchantName: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchantLogoUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchantBgUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offerImageUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    listviewColor: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    detailviewColor: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    validity: {
        allowNull: true,
        type: DataTypes.DATE,
    },

    applicability: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    couponCode: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    conditions: {
        allowNull: true,
        type: DataTypes.ARRAY( DataTypes.TEXT ), // eslint-disable-line new-cap
    },

    redemptionLink: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    mainTnc: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
