// @infino/data-models/lib/postgres/schema/curated-giftcard.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    collectionId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'collections',
            key: 'id',
        },
    },

    giftcardId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'giftcards',
            key: 'id',
        },
    },

    merchantLogoUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    category: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offerDiscount: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    listviewColor: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offerImageUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
