// @infino/data-models/lib/postgres/schema/finance.js
const { Sequelize, DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: Sequelize.UUIDV4
    },

    name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    bankId: {
        allowNull: true,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'bank',
            key: 'id'
        }
    },

    logoUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    color: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    type: {
        allowNull: false,
        type: DataTypes.ENUM,
        values: [ 'BANK', 'FINTECH', 'NETWORK' ]
    },

    quickFilter: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    hostName: {
        allowNull: false,
        type: DataTypes.STRING
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    }

};
