// @infino/data-models/lib/postgres/schema/integration.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    templateId: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    logoUrl: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    hostedObjectKey: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    type: {
        type: DataTypes.ENUM( // eslint-disable-line new-cap
            'FINTECH',
            'BANK',
            'PAYMENT_NETWORK',
        ),
        allowNull: false,
        defaultValue: 'FINTECH',
    },

    color: {
        type: DataTypes.STRING,
        allowNull: false,
        defaultValue: '#FFFFFF',
    },

    description: {
        type: DataTypes.STRING,
        allowNull: false,
        defaultValue: 'Offers are applicable on all Visa Cards.',
    },

    hostName: {
        type: DataTypes.STRING,
        allowNull: false,
        defaultValue: 'olamoney.com',
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
