// @infino/data-models/lib/postgres/schema/mobikwik-payment-response.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    orderId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'orders',
            key: 'id',
        },
    },

    responseCode: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    isEmailDispatched: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    gatewayTransactionId: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    gatewayTransactionTime: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
