// @infino/data-models/lib/postgres/schema/online-offer.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    merchantId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'online_merchants',
            key: 'id'
        }
    },

    offerText: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    description: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    validity: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    applicability: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    couponCode: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    conditions: {
        allowNull: true,
        type: DataTypes.ARRAY( DataTypes.TEXT ), // eslint-disable-line new-cap
    },

    conditionUrl: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    infinoConditions: {
        allowNull: true,
        type: DataTypes.ARRAY( DataTypes.TEXT ), // eslint-disable-line new-cap
    },

    redemptionLink: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    activeStatus: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: true,
    },

    offerColor: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
