// @infino/data-models/lib/postgres/schema/purchased-giftcard.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    paymentResponseReference: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'mobikwik_payment_responses',
            key: 'id',
        },
    },

    providerReference: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    userId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'users',
            key: 'id',
        },
    },

    cardNumber: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    cardPin: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    activationUrl: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    activationCode: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    zkCardReference: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    expiresAt: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
