// @infino/data-models/lib/postgres/schema/qr-auth-request.js
const { Sequelize, DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: Sequelize.UUIDV4
    },

    authenticatorSessionId: {
        allowNull: true,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'sessions',
            key: 'id',
        },
        defaultValue: null
    },

    authenticatedSessionId: {
        allowNull: true,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'sessions',
            key: 'id',
        },
        defaultValue: null
    },

    isUsed: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    isAuthenticated: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    nonce: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    expiresAt: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    }

};
