// @infino/data-models/lib/postgres/schema/user-online-offer.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    onlineOfferId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'online_offers',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    userId: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'users',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    timestamp: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    redeemed: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },
};
