// @infino/data-models/lib/postgres/schema/vault-request-metadata.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    uniqueId: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    creationIpAddress: {
        allowNull: false,
        type: DataTypes.CIDR,
    },

    consumptionIpAddress: {
        allowNull: true,
        type: DataTypes.CIDR,
    },

    isError: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    isSuccess: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: true,
    },

    errorReason: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    errorDetails: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    createdAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updatedAt: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
