// @infino/data-models/ancillary/migrations/20200607050552-create-user-saving.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    up: queryInterface => queryInterface.createTable( 'user_savings', {

        id: {
            type: DataTypes.UUID,
            allowNull: false,
            primaryKey: true,
            defaultValue: DataTypes.UUIDV4,
        },

        user_id: {
            type: DataTypes.UUID,
            allowNull: false,
            references: {
                model: 'users',
                key: 'id',
                onDelete: 'CASCADE',
            },
        },

        saving_amount: {
            type: DataTypes.FLOAT,
            allowNull: false,
        },

        saving_type: {
            type: DataTypes.ENUM( 'GIFTCARD', 'COUPON' ), // eslint-disable-line new-cap
            allowNull: false,
        },

        is_confirmed_saving: {
            type: DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: true,
        },

        created_at: {
            allowNull: false,
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },

        updated_at: {
            allowNull: false,
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },

    } ),

    down: queryInterface => queryInterface.dropTable( 'user_savings' )

};
