// @infino/data-models/ancillary/migrations/20200420180700-finance.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    up: queryInterface => queryInterface.createTable( 'finances', {

        id: {
            allowNull: false,
            primaryKey: true,
            type: DataTypes.UUID,
            defaultValue: DataTypes.UUIDV4
        },

        name: {
            allowNull: false,
            type: DataTypes.STRING,
        },

        bank_id: {
            allowNull: true,
            type: DataTypes.UUID,
            references: {
                onDelete: 'CASCADE',
                model: 'banks',
                key: 'id'
            }
        },

        logo_url: {
            allowNull: false,
            type: DataTypes.STRING,
        },

        color: {
            allowNull: false,
            type: DataTypes.STRING,
        },

        type: {
            allowNull: false,
            type: DataTypes.ENUM,
            values: [ 'BANK', 'FINTECH', 'NETWORK' ]
        },

        quick_filter: {
            allowNull: false,
            type: DataTypes.BOOLEAN,
            defaultValue: false,
        },

        host_name: {
            allowNull: false,
            type: DataTypes.STRING
        },

        created_at: {
            allowNull: false,
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },

        updated_at: {
            allowNull: false,
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        }

    } ),
    down: queryInterface => queryInterface.dropTable( 'finances' )

};
