// @infino/data-models/ancillary/migrations/20200606081427-online-offer-finance.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    up: queryInterface => queryInterface.createTable( 'online_offer_finances', {

        id: {
            allowNull: false,
            primaryKey: true,
            type: DataTypes.UUID,
            defaultValue: DataTypes.UUIDV4,
        },

        offer_id: {
            allowNull: false,
            type: DataTypes.UUID,
            references: {
                model: 'online_offers',
                key: 'id'
            }
        },

        finance_id: {
            allowNull: true,
            type: DataTypes.UUID,
            references: {
                onDelete: 'CASCADE',
                model: 'finances',
                key: 'id',
            },
        },

        created_at: {
            allowNull: false,
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },

        updated_at: {
            allowNull: false,
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },

    } ),
    down: queryInterface => queryInterface.dropTable( 'online_offer_finances' )

};
