// @infino/data-models/ancillary/schema/card.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    card_type_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'card_types',
            key: 'id',
        },
    },

    user_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'users',
            key: 'id',
        },
    },

    name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    first_six: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    last_four: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    card_token: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    is_active: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: true,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
