// @infino/data-models/ancillary/schema/card-retrieval-request.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    card_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'cards',
            key: 'id',
        },
    },

    session_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'sessions',
            key: 'id',
        },
    },

    vault_request_metadata_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'vault_request_metadata',
            key: 'id',
        },
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
