// @infino/data-models/ancillary/schema/curated-offer.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    collection_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'collections',
            key: 'id',
        },
    },

    category: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    offer_discount: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    offer_title: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    description: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchant_name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchant_logo_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchant_bg_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offer_image_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    listview_color: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    detailview_color: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    validity: {
        allowNull: true,
        type: DataTypes.DATE,
    },

    applicability: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    coupon_code: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    conditions: {
        allowNull: true,
        type: DataTypes.ARRAY( DataTypes.TEXT ), // eslint-disable-line new-cap
    },

    redemption_link: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    main_tnc: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
