// @infino/data-models/ancillary/schema/curated-giftcard.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    collection_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'collections',
            key: 'id',
        },
    },

    giftcard_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'giftcards',
            key: 'id',
        },
    },

    merchant_logo_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    category: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offer_discount: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    listview_color: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offer_image_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
