// @infino/data-models/ancillary/schema/giftcard.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    provider_id: {
        allowNull: false,
        type: DataTypes.UUID,
        onDelete: 'CASCADE',
        references: {
            model: 'providers',
            key: 'id',
        }
    },

    merchant_id: {
        allowNull: false,
        type: DataTypes.UUID,
        onDelete: 'CASCADE',
        references: {
            model: 'online_merchants',
            key: 'id',
        }
    },

    giftcard_name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    max_clubbable_quantity: {
        allowNull: false,
        type: DataTypes.INTEGER,
    },

    base_commission: {
        allowNull: false,
        type: DataTypes.FLOAT
    },

    selling_rate: {
        allowNull: false,
        type: DataTypes.FLOAT
    },

    about: {
        allowNull: true,
        type: DataTypes.STRING
    },

    expiry: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    redemption_instructions: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    applicability: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    redemption_link: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    sale_applicability: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    exclusive_online: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    exclusive_offline: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    tnc_url: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    denomination_type: {
        allowNull: false,
        type: DataTypes.ENUM,
        values: [ 'RANGE', 'SLAB', 'GYFTR_SLAB' ]
    },

    validity: {
        allowNull: true,
        type: DataTypes.STRING
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
    }

};
