// @infino/data-models/ancillary/schema/online-offer.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    merchant_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'online_merchants',
            key: 'id'
        }
    },

    offer_text: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    description: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    validity: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    applicability: {
        allowNull: false,
        type: DataTypes.TEXT,
    },

    coupon_code: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    conditions: {
        allowNull: true,
        type: DataTypes.ARRAY( DataTypes.TEXT ), // eslint-disable-line new-cap
    },

    condition_url: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    infino_conditions: {
        allowNull: true,
        type: DataTypes.ARRAY( DataTypes.TEXT ), // eslint-disable-line new-cap
    },

    redemption_link: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    offer_color: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
