// @infino/data-models/ancillary/schema/online-merchant.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    merchant_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    logo_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    cover_image_url: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    android_package_identifier: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    ios_package_identifier: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    host_name: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    background_color: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
