// @infino/data-models/ancillary/schema/order.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    user_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'users',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    payment_method_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'payment_methods',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    provider_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'providers',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    giftcard_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'giftcards',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    amount: {
        allowNull: false,
        type: DataTypes.FLOAT,
    },

    profit_amount: {
        allowNull: false,
        type: DataTypes.FLOAT,
    },

    profit_percentage: {
        allowNull: false,
        type: DataTypes.FLOAT,
    },

    payable_amount: {
        allowNull: false,
        type: DataTypes.FLOAT,
    },

    is_consumed: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
