// @infino/data-models/ancillary/schema/purchased-giftcard.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    payment_response_reference: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'mobikwik_payment_responses',
            key: 'id',
        },
    },

    provider_reference: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    user_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'users',
            key: 'id',
        },
    },

    card_number: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    card_pin: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    activation_url: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    activation_code: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    zk_card_reference: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    expires_at: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
