// @infino/data-models/ancillary/schema/qr-auth-request.js
const { Sequelize, DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: Sequelize.UUIDV4
    },

    authenticator_session_id: {
        allowNull: true,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'sessions',
            key: 'id',
        },
        defaultValue: null
    },

    authenticated_session_id: {
        allowNull: true,
        type: DataTypes.UUID,
        references: {
            onDelete: 'CASCADE',
            model: 'sessions',
            key: 'id',
        },
        defaultValue: null
    },

    is_used: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    is_authenticated: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    nonce: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    expires_at: {
        allowNull: false,
        type: DataTypes.DATE,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    }

};
