// @infino/data-models/ancillary/schema/user-giftcard.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    giftcard_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'giftcards',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    user_id: {
        allowNull: false,
        type: DataTypes.UUID,
        references: {
            model: 'users',
            key: 'id',
            onDelete: 'CASCADE',
        }
    },

    timestamp: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    redeemed: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },
};
