// @infino/data-models/ancillary/schema/vault-request-metadata.js
const { DataTypes } = require( 'sequelize' );

module.exports = {

    id: {
        allowNull: false,
        primaryKey: true,
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
    },

    unique_id: {
        allowNull: false,
        type: DataTypes.STRING,
    },

    creation_ip_address: {
        allowNull: false,
        type: DataTypes.CIDR,
    },

    consumption_ip_address: {
        allowNull: true,
        type: DataTypes.CIDR,
    },

    is_error: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: false,
    },

    is_success: {
        allowNull: false,
        type: DataTypes.BOOLEAN,
        defaultValue: true,
    },

    error_reason: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    error_details: {
        allowNull: true,
        type: DataTypes.STRING,
    },

    created_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

    updated_at: {
        allowNull: false,
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
    },

};
